File: 2021-g10/phomai/phomai.cpp
#include <iostream>
#include <fstream>
#include <vector>

#define inputFile "phomai.inp"
#define outputFile "phomai.out"

using namespace std;

int a, b; // khối lượng 2 miếng phô mai


void Input()
{
    ifstream f;
    f.open(inputFile);

    f >> a >> b;

    f.close();
}


// Tìm ước chung lớn nhất
int GCD(int m, int n)
{
    while (n != 0)
    {
        int t = n;
        n = m % n;
        m = t;
    }

    return m;
}


int Process()
{
    if (a == b) return 0;

    int c = 0; // count: số lần ăn phô mai của cáo

    // Tìm ước chung lớn nhất của a và b
    int gcd = GCD(a, b);

    // Lần lượt chia a cho 2, 3 và 5 để đếm số lần cắn phô mai của cáo
    vector<int> v = {2, 3, 5};

    for (int i : v)
    {
        while (a % i == 0)
        {
            // Nếu đã bằng UCLN thì không cần cắn miếng phô mai a nữa
            if (a == gcd) break;
            a = a / i;
            c++;
        }
    }

    // Nếu a vẫn chưa chạm được tới UCLN thì không thể chia phô mai được
    if (a != gcd) return -1;

    // Lần lượt chia b cho 2, 3 và 5 để đếm số lần cắn phô mai của cáo
    for (int i : v)
    {
        while (b % i == 0)
        {
            // Nếu đã bằng UCLN thì không cần cắn miếng phô mai a nữa
            if (b == gcd) break;
            b = b / i;
            c++;
        }
    }

    // Nếu b vẫn chưa chạm được tới UCLN thì không thể chia phô mai được
    if (b == gcd)
        return c;
    else
        return -1;
}


void Output()
{
    int res = Process();

    ofstream f;
    f.open(outputFile);

    f << res;

    f.close();
}


int main()
{
    Input();
    Output();

    return 0;
}

File: 2021-g10/phomai/phomai.py
import os
import sys

input_file = os.path.join(sys.path[0], 'phomai.inp')
output_file = os.path.join(sys.path[0], 'phomai.out')

a = b = 0 # khối lượng 2 miếng phô mai


def input_data():
    with open(input_file) as f:
        global a, b
        a, b = map(int, f.readline().split())


# Tìm ước chung lớn nhất
def gcd(m, n):
    while n != 0:
        m, n = n, m % n

    return m


def process():
    global a, b
    if a == b:
        return 0

    c = 0 # count: số lần ăn phô mai của cáo
  
    # Tìm ước chung lớn nhất của a và b
    g = gcd(a, b)

    # Lần lượt chia a cho 2, 3 và 5 để đếm số lần cắn phô mai của cáo
    for i in [2, 3, 5]:
        while a % i == 0:
            # Nếu đã bằng UCLN thì không cần cắn miếng phô mai a nữa
            if a == g:
                break
            a = a / i
            c += 1

    # Nếu a vẫn chưa chạm được tới UCLN thì không thể chia phô mai được
    if a != g:
        return -1 

    # Lần lượt chia b cho 2, 3 và 5 để đếm số lần cắn phô mai của cáo
    for i in [2, 3, 5]:
        while b % i == 0:
            # Nếu đã bằng UCLN thì không cần cắn miếng phô mai a nữa
            if b == g:
                break
            b = b / i
            c += 1 

    # Nếu b vẫn chưa chạm được tới UCLN thì không thể chia phô mai được
    if b == g:
        return c
    else:
        return -1


def output():
    res = str(process())

    with open(output_file, 'w') as f:
        f.write(res)


if __name__ == '__main__':
    input_data()
    output()